src/fhirtypes/ElementDefinition.ts

```diff
@@ -269,6 +271,40 @@ export class ElementDefinition {
       throw new WideningCardinalityError(this.min, this.max, min, max);
     }
 
+    // Sliced elements and slices have special card rules described here:
+    // http://www.hl7.org/fhiR/profiling.html#slice-cardinality
+    // If element is slice definition
+    if (this.slicing) {
+      const slices = this.structDef.elements.filter(e => e.id !== this.id && e.path === this.path);
+      const sumOfMins = slices.reduce((prev, curr) => (prev += curr.min), 0);
+      // Check that new max >= sum of mins of children
+      if (!isUnbounded && sumOfMins > maxInt) {
+        throw new InvalidSumOfSliceMinsError(sumOfMins, max);
```

Comment:
For this error and the other cases where `InvalidSumOfSliceMinsError` and `InvalidMaxOfSliceError` are thrown in this method, I think the message should include the name of the element being sliced. Also, if a specific slice's cardinality caused the error to be thrown, that element's name should also be included in the message.

Reply:
Oh yeah good point. I think since the `InvalidMaxOfSliceError` is only when we are constraining cardinality on the element being sliced, we will have information on the sliced element from the `sourceInfo` correct? But I do think that error should list the name of the slice that has too large a maximum. And then since the `InvalidSumOfSliceMinsError` can happen either when we are constraining an element being sliced, or constraining a slice, I agree we should put the name of the element being sliced in the error since we may not have `sourceInfo` on it.